src/server/protocol.ts

```diff
@@ -1829,6 +1830,38 @@ namespace ts.server.protocol {
         closedFiles?: string[];
     }
 
+    export interface UpdateFileSystemRequest extends Request {
+        command: CommandTypes.UpdateFileSystem;
+        arguments: UpdateFileSystemRequestArgs;
+    }
+
+    /**
+     * Arguments to UpdateFileSystemRequest. Tracks changes to non-opened files.
+     */
+    export interface UpdateFileSystemRequestArgs {
+        /** For now, only 'memfs', initially for exclusive in-memory operation, but it could be other in-memory names later */
+        fileSystem: string;
+        /** List of newly created or newly available files. */
+        created: FileSystemRequestArgs[];
+        /** Names of just-deleted files. */
+        deleted: string[];
+        /** List of updated files. */
+        updated: FileSystemRequestArgs[];
+    }
+
+    export interface FileSystemRequestArgs extends FileRequestArgs {

```

Comment:
Presumably, it should resemble `OpenRequestArgs`.

Reply:
Yeah, that's where I copied `scriptKindName` from too. Still, since we're imitating a filesystem, I don't think projectFileName (from FileRequestArgs) or scriptKindName make sense.